File: examples/java-unittest/MyCounter.java
import java.util.Random;

class MyCounter {
    public static boolean faulty = false;

    private Random r = new Random();

    public MyCounter() {
	if (!faulty) {
	    value = 0;
	} else {
	    value = r.nextInt();
	}
    }

    public void inc() {
	value++;
    }

    public void reset() {
	value = 0;
    }

    public int count() {
	return value;
    }

    private int value;
};

File: examples/shortener/mycounter.cc
#include "mycounter.h"

int MyCounter::counter=0;

MyCounter::MyCounter()
{
  if ((counter%7)!=0 || counter==0)
    value = 0;
  counter ++;
}

void MyCounter::inc() {
    value += 1;
}

void MyCounter::reset() {
    value = 0;
}

int MyCounter::count() {
    return value;
}
